### ISSUE TITLE
find-view not finding views

### ISSUE THREAD

**Issue by kenrestivo** [clojure-android/neko#58]
```clojure
  (debug/safe-for-ui
   (view/find-view (debug/*a) ::playing-button))
```
returns nil

but the view that is set at onCreate time definitely has an element with that id, and I can see it displayed. When run from inside the program (rather than the repl), using this instead of (debug/*a), it also returns nil.

Interestingly, in attempting to debug this, I tried this from a REPL:

```clojure
  (debug/safe-for-ui
   (activity/get-decor-view (debug/*a)))
```

And got a strange 
```clojure
#object[com.android.internal.policy.impl.PhoneWindow$DecorView 0xa104b800 "com.android.internal.policy.impl.PhoneWindow$DecorView@a104b800"]
```

returned instead of my view.

I'm not sure what's going on here.

**Comment by alexander-yakushev** [comment]
That's a correct DecorView object you are getting. Can you call (`.getTag)` on it?

**Comment by kenrestivo** [comment]
.getTag returns nil.

**Comment by alexander-yakushev** [comment]
Yes, something's fishy going on. Can you please show the code of your onCreate? Also, are you sure `(debug/*a)` returns the very activity you think it should return?

**Comment by kenrestivo** [comment]
Gladly: https://github.com/kenrestivo/spazradioapp/blob/new-neko/src/clojure/org/spaz/radio/player.clj#L201

Thanks for your help!

**Comment by alexander-yakushev** [comment]
Sorry, I totally forgot. To enable this behavior (finding views by `:id` directly in their activity) you shouldn't call this anymore:

`(set-content-view! activity (make-ui activity ui-tree))`

But instead:

`(set-content-view! activity ui-tree)`

In other words, you don't have to call `make-ui` explicitly anymore, set-content-view! does it automatically now.

**Comment by alexander-yakushev** [comment]
Also, if you use Emacs, you can add this to your `init.el`:

```scheme
  (add-hook 'clojure-mode-hook
            #'(lambda ()
                (put 'defactivity 'clojure-backtracking-indent '(4 (2)))))
```

This will make indentation in defactivity nice again.

**Comment by kenrestivo** [comment]
Pilot error! I don't need to make-ui before handing this to set-content-view!

**Comment by alexander-yakushev** [comment]
You are welcome! Glad it works now.

**Comment by kenrestivo** [comment]
Um, I spoke too soon. .getTag works, but find-view does not still

```clojure
  (debug/safe-for-ui
   (view/find-view (debug/*a) ::playing-text))
```

=> nil

**Comment by kenrestivo** [comment]
Also returns nil: 
```clojure
(debug/safe-for-ui
   (-> (debug/*a)
       activity/get-decor-view
       .getTag
       (get ::playing-text)))
````
=> nil

the get-decor view is in fact returning the correct layout, I can see the :id

**Comment by alexander-yakushev** [comment]
Heh, I see the problem now. It is all those `:id-holder`s you have in your layout which are rendundant. Whenever you set `:id-holder` attribute for a container, it means that all its internal elements will be caught by him and stored in its tag. It is useful when you have a listview and each element is compound - so you keep references to all individual pieces inside each element container.

Now to set-content-view!. It implicitly puts :id-holder on top-level DecorView, so all those :id's propagate all the way up to it — unless someone caughts them in the middle, as in your case.

**TL;DR: Remove `:id-holder`s from your layout and you'll be fine :).**

**Comment by kenrestivo** [comment]
That's it! Thanks for making the whole API simpler. I
